components/infra-proxy-service/service/service.go

```diff
@@ -41,13 +43,17 @@ func Start(l logger.Logger, migrationsConfig migration.Config, connFactory *secu
 }
 
 // ParseStorageError parses common storage errors into a user-readable format.
-func ParseStorageError(err error, id interface{}, noun string) error {
+func ParseStorageError(err error, v interface{}, noun string) error {
 	if err != nil {
 		switch err {
 		case storage.ErrNotFound:
-			return status.Errorf(codes.NotFound, "no %s found with id %q", noun, id)
+			return status.Errorf(codes.NotFound, "no %s found with ID %q", noun, reflect.Indirect(reflect.ValueOf(v)).FieldByName("Id"))
+		case storage.ErrCannotDelete:
+			return status.Errorf(codes.FailedPrecondition, "cannot delete server %q because it still has organizations attached", reflect.Indirect(reflect.ValueOf(v)).FieldByName("Id"))
```

Comment:
I am not 100% sure about the error code here but I thought it would be fitted for our need.

Reply:
Receive the request object reference in the interface and extract the value base on the need.